apps/cli/src/tui/app.tsx

```
@@ -467,3 +488,58 @@ export function App({
+// result can legitimately contain a literal `*`/`#`/backtick that must render as-is, not get parsed
+// as markdown syntax.
+// Its own memoized component, not an inline `.map()` in App's own JSX: `state.transcript`'s
+// reference only changes on an actual append (state/reducer.ts), so `memo` here lets React skip
+// rebuilding and re-diffing the whole elements array on a render triggered by unrelated state (a
+// streamed token's `state.turn.tokens` tick, a scroll-banner flip) — not just skip the per-row
+// markdown work `TranscriptRow`'s own `memo` (below) already bails out of.
+const TranscriptList = memo(function TranscriptList({ transcript }: { transcript: TranscriptEntry[] }) {
+  return (
+    <>
+      {transcript.map((entry, index) => (
+        <TranscriptRow key={index} entry={entry} />
```

Biome (2.5.7)

[error] Avoid using the index of an array as key property in an element.

(lint/suspicious/noArrayIndexKey)